Controller/Payment/Redirect.php

```diff
@@ -4,26 +4,32 @@
 
 use Exception;
 use Magento\Framework\Controller\ResultFactory;
-use Midtrans\Snap\Gateway\SnapApi;
 use Midtrans\Snap\Gateway\Config\Config;
+use Midtrans\Snap\Gateway\SnapApi;
 
 class Redirect extends AbstractAction
 {
     public function execute()
     {
-        $paymentCode = $this->getCode();
-        $requestConfig = $this->getData()->getRequestConfig($paymentCode);
-        $enableRedirect = $this->getData()->isRedirect();
+        try {
+            $paymentCode = $this->getCode();
+            $requestConfig = $this->getData()->getRequestConfig($paymentCode);
+            $enableRedirect = $this->getData()->isRedirect();
 
-        $payloads = $this->getPayload($requestConfig);
-        $is3ds = $requestConfig['is3ds'];
+            $payloads = $this->getPayload($requestConfig);
+            $is3ds = $requestConfig['is3ds'];
+            $isProduction = $this->getData()->isProduction();
 
-        Config::$is3ds = $is3ds;
-        Config::$isProduction = $this->getData()->isProduction();
-        Config::$serverKey = $this->getData()->getServerKey($paymentCode);
-        Config::$isSanitized = false;
+            Config::$isProduction = $isProduction;
+            Config::$serverKey = $this->getData()->getServerKey($paymentCode);
+            Config::$isSanitized = true;
+            Config::$is3ds = $is3ds;
+
+            /*Override notification, if override notification from admin setting is active (default is active) */
+            if ($this->getData()->isOverrideNotification()) {
+                Config::$overrideNotifUrl = $this->getData()->getNotificationEndpoint();
```

Comment:
Better check first the value of `getNotificationEndpoint()`, if it `null` then do not set `Config::$overrideNotifUrl`

Reply:
done, handle with this
`
if ($this->dataConfig->isOverrideNotification() && $this->dataConfig->getNotificationEndpoint() != null) {
            Config::$overrideNotifUrl = $this->dataConfig->getNotificationEndpoint();
        }
`